app/dashboard/views.py

```diff
@@ -1133,9 +1139,111 @@ def profile(request, handle):
                 },
             },
         }
+        return TemplateResponse(request, 'profiles/profile.html', context, status=status)
+
+    context['preferred_payout_address'] = profile.preferred_payout_address
+    order_by = request.GET.get('order_by', '-modified_on')
+    if context['preferred_payout_address']:
+        owned_kudos = Token.objects.filter(
+            owner_address=to_checksum_address(context['preferred_payout_address']),
+            contract__network=settings.KUDOS_NETWORK
```

Comment:
We should `select_related` on the `'contract'` since we're filtering on values from it here.